app/gosqs/send_message_test.go

```diff
@@ -0,0 +1,219 @@
+	status, _ := SendMessageV1(r)
+
+	// Check the queue
+	assert.Equal(t, 1, len(q.Messages))
+	// Check the response
+	assert.Equal(t, http.StatusOK, status)
+
+	// Send the same message (have DeduplicationId)
+	status, _ = SendMessageV1(r)
+	// Response is "success"
+	assert.Equal(t, http.StatusOK, status)
+	// Only 1 message should be in the queue
+	assert.Equal(t, 1, len(q.Messages))
+}
+
+func TestSendMessageV1_request_transformer_error(t *testing.T) {
+	app.CurrentEnvironment = fixtures.LOCAL_ENVIRONMENT
+	defer func() {
+		utils.ResetApp()
+		utils.REQUEST_TRANSFORMER = utils.TransformRequest
+	}()
+
+	utils.REQUEST_TRANSFORMER = func(resultingStruct interfaces.AbstractRequestBody, req *http.Request) (success bool) {
+		return false
+	}
+
+	_, r := utils.GenerateRequestInfo("POST", "/", nil, true)
+	code, _ := SendMessageV1(r)
+
+	assert.Equal(t, http.StatusBadRequest, code)
+}
+
+func TestSendMessageV1_MaximumMessageSize_MessageTooBig(t *testing.T) {
+	app.CurrentEnvironment = fixtures.LOCAL_ENVIRONMENT
+	defer func() {
+		utils.ResetApp()
+		utils.REQUEST_TRANSFORMER = utils.TransformRequest
+	}()
+
+	sendMessageRequest_success := models.SendMessageRequest{
+		QueueUrl:    "http://localhost:4200/new-queue-1",
+		MessageBody: "Test Message",
+	}
+	utils.REQUEST_TRANSFORMER = func(resultingStruct interfaces.AbstractRequestBody, req *http.Request) (success bool) {
+		v := resultingStruct.(*models.SendMessageRequest)
+		*v = sendMessageRequest_success
+		return true
+	}
+
+	q := &app.Queue{
+		Name:               "new-queue-1",
+		MaximumMessageSize: 1,
+	}
+	app.SyncQueues.Queues["new-queue-1"] = q
+
+	_, r := utils.GenerateRequestInfo("POST", "/", nil, true)
+	status, response := SendMessageV1(r)
+
+	// Check the response
+	assert.Equal(t, http.StatusBadRequest, status)
+	errorResponse, ok := response.(models.ErrorResponse)
+	assert.True(t, ok)
+	assert.Equal(t, "MessageTooBig", errorResponse.Result.Type)
+}
+
+func TestSendMessageV1_POST_QueueNonExistant(t *testing.T) {
+	app.CurrentEnvironment = fixtures.LOCAL_ENVIRONMENT
+	defer func() {
+		utils.ResetApp()
+		utils.REQUEST_TRANSFORMER = utils.TransformRequest
+	}()
+
+	sendMessageRequest_success := models.SendMessageRequest{
+		QueueUrl:    "http://localhost:4200/new-queue-1",
+		MessageBody: "Test Message",
+	}
+	utils.REQUEST_TRANSFORMER = func(resultingStruct interfaces.AbstractRequestBody, req *http.Request) (success bool) {
+		v := resultingStruct.(*models.SendMessageRequest)
+		*v = sendMessageRequest_success
+		return true
+	}
+
+	// No test queue is added to app.SyncQueues
+
+	_, r := utils.GenerateRequestInfo("POST", "/", nil, true)
+	status, response := SendMessageV1(r)
+
+	// Check the status code is what we expect.
+	assert.Equal(t, http.StatusBadRequest, status)
+
+	// Check the response body is what we expect.
+	errorResponse, ok := response.(models.ErrorResponse)
+	assert.True(t, ok)
+	assert.Equal(t, "Not Found", errorResponse.Result.Type)
+}
```

Comment:
These look great!  Optional: Do we want to do maybe just 1 more test with some attributes?

Reply:
Sorry, what test would you like to add? For MessageAttributes? Or is it any of Queue Attribute?